app/gui2/src/components/visualizations/WarningsVisualization.vue

```diff
@@ -9,21 +9,39 @@ export const defaultPreprocessor = [
 </script>
 
 <script setup lang="ts">
-import { VisualizationContainer } from '@/util/visualizationBuiltins'
+import SvgIcon from '@/components/SvgIcon.vue'
+import { Pattern } from '@/util/ast/match'
+import { useVisualizationConfig, VisualizationContainer } from '@/util/visualizationBuiltins'
 
 type Data = string[]
 
 const props = defineProps<{ data: Data }>()
+
+const config = useVisualizationConfig()
+
+const removeWarnings = Pattern.parse('__.remove_warnings')
```

Comment:
should this be in a non-`setup` block? (so that it's only ever declared once globally?)

Reply:
also, i think maybe `Pattern.parse` should be exposed as part of the builtins? or perhaps `content` should (also?) accept a string

Reply:
> should this be in a non-`setup` block? (so that it's only ever declared once globally?)

Definitely, good catch.

> also, i think maybe `Pattern.parse` should be exposed as part of the builtins? or perhaps `content` should (also?) accept a string

Do you mean exposed through `useVisualizationConfig`? `Pattern`s are not a visualization or even `GraphEditor` concept; they are a way of talking about structure of ASTs. I think modules that work with them can import them like other `Ast` APIs.

I would expect a `content: string` to be static code. Is it not clearer to pass a `Pattern` than a string intended to be interpreted as a `Pattern`? Also, not every pattern is built from a string--#9409 introduces a type-safe means of composing patterns at runtime.

Reply:
> Do you mean exposed through `useVisualizationConfig`? 

nah, i mean `'@/util/visualizationBuiltins'` - i think ideally this functionality would be exposed to custom visualizations - but of course there's nothing wrong with disallowing it for now, until we're sure this is the api we want to expose publicly